cilium/CFP-1189-connectivity-tests-speedup.md

```diff
@@ -0,0 +1,52 @@
+# CFP-1189: Cilium CLI connectivity tests speedup
+
+**SIG: SIG-USER**
+
+**Begin Design Discussion:** 2024-01-19
+
+**Cilium Release:** 1.15
+
+**Authors:** Viktor Kurchenko <viktor.kurchenko@isovalent.com>
+
+## Summary
+
+The CFP describes a new approach to run Cilium connectivity tests.
+The idea is to group all the tests in small sets and run them in parallel.
+
+## Motivation
+
+Currently, connectivity tests might run in CI for over 1 hour (it depends
+on many factors) and the test case count constantly increasing.
+To make CI pipelines faster and cheaper we can consider the connectivity
+tests parallelization approach.
+
+## Goals
+
+* Group connectivity tests into independent sets that can be run concurrently.
+* Run each independent test set concurrently (all together or in batches).
+* Collect test results and display them periodically.
```

Comment:
> It looks like a different subject to me and it may make sense to have the CFP focused on test parallelisation.

Maybe but if some tests are run in parallel how it will look from the user's perspective?
Output might be unreadable, isn't it?

Reply:
I am not sure to follow. I have looked at a [simple example](https://github.com/cilium/cilium-cli/blob/main/connectivity/tests/pod.go#L17). The curl test should be free of side effect. The flow validation, I am guessing so and for metrics validation I don't know.
Currently tests are run sequentially in the order they have been registered. They are run in a separate go routine but the loop read from a channel populated at the completion of a test case before going on with the next one:
https://github.com/cilium/cilium-cli/blob/main/connectivity/check/context.go#L402-L433
If we move the channel read outside of the loop we can still collect the results and present them in an ordered way. Is it what you mean? Or am I not getting your point?

Reply:
Yes, that is what I meant.